src/VirtualClient/VirtualClient.Dependencies/BufferTimeWaiter.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Dependencies
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Polly;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// Provides functionality to wait for given time..
+    /// </summary>
+    public class BufferTimeWaiter : VirtualClientComponent
```

Comment:
Thanks for adding this. Let's just call it WaitExecutor, and let's add a minimum amount of document for this.